### DISCUSSION TITLE
20230708 周会纪要

### DISCUSSION THREAD

**Discussion by maronghe** [arana-db/arana#699] [Weekly meeting]
## 1. 本周进展

| 功能点 | 参与人员 | 本周工作进展 | 备注 |
| --- | --- | --- | --- |
| [Feature] add log for sharding logic SQL and real SQL. | @dongjh | 新 PR 从 context 取 traceId：1、log 中添加 trace id  2、添加 W3C trace context。 | [https://github.com/arana-db/arana/issues/596](https://github.com/arana-db/arana/issues/596) |
| 1、[TX] support transaction XA  @chunshao  <br> ~~2、[XA] build auto clean transaction log ability @pangx~~ <br> ~~3、[XA] build transaction log ability @liy~~ <br> 4、[XA] build XA transaction back-and-forth decision-making  @haohf <br> 5、[tx] add deadlock for arana transaction. @liuyc  | @chunshao <br> @liuyc <br> @haohf <br> @liy（finish）<br> @pangx（finish） |  | [https://github.com/arana-db/arana/issues/324](https://github.com/arana-db/arana/issues/324) <br> [https://github.com/arana-db/arana/issues/556](https://github.com/arana-db/arana/issues/556) <br> [https://github.com/arana-db/arana/issues/555](https://github.com/arana-db/arana/issues/555) <br> [https://github.com/arana-db/arana/issues/602](https://github.com/arana-db/arana/issues/602) <br> [https://github.com/arana-db/arana/issues/607](https://github.com/arana-db/arana/issues/607) |
| [kernel] run arana at the envoy. | @haohf |  | [https://github.com/arana-db/arana/issues/649](https://github.com/arana-db/arana/issues/649) |
| Support multiple columns complex sharding key. | @sanxun0325 @sudu | 补充 UT[https://github.com/arana-db/arana/pull/681](https://github.com/arana-db/arana/pull/681) | [https://github.com/arana-db/arana/issues/425](https://github.com/arana-db/arana/issues/425) |
| [Feature] Support SQL join | @sudu @huangwk |  | [https://github.com/arana-db/arana/issues/600](https://github.com/arana-db/arana/issues/600) <br> [https://github.com/arana-db/arana/issues/622](https://github.com/arana-db/arana/issues/622) |
| Implement a low-level HASH-JOIN dataset | @wangrui @sudu | 补充UT内容，有比较大改动，重新 review 代码[https://github.com/arana-db/arana/pull/668](https://github.com/arana-db/arana/pull/668) | [https://github.com/arana-db/arana/issues/642](https://github.com/arana-db/arana/issues/642) |
| [Bug] fix SHOW INDEXES FROM xxx result error | @pangx | | [https://github.com/arana-db/arana/issues/678](https://github.com/arana-db/arana/issues/678) |
| ~~[Bug] select from table not exist cause panic~~ | @gongna | 新 PR 已提交[https://github.com/arana-db/arana/pull/693](https://github.com/arana-db/arana/pull/693) | [https://github.com/arana-db/arana/issues/660](https://github.com/arana-db/arana/issues/660) |
| [Bug] fix show tables xxx statement error.  | @zhangxy | 会后私聊 | [https://github.com/arana-db/arana/issues/677](https://github.com/arana-db/arana/issues/677) |
| ~~[CI] Integrated code check tools~~ | @weideman |  | [https://github.com/arana-db/arana/issues/670](https://github.com/arana-db/arana/issues/670) |
| MySQL  window 函数实现 | @yangwei | ~~聚集函数~~ & 非聚集函、数补充测试 case，first/last 场景补充 case | [https://github.com/arana-db/arana/issues/667](https://github.com/arana-db/arana/issues/667) |
| [SQL] Support MySQL Prepare SQL | @maronghe | 正在调研 https://github.com/arana-db/arana/discussions/700 | [https://github.com/arana-db/arana/issues/591](https://github.com/arana-db/arana/issues/591) |
| [SQL] Explain | @maorui | [https://github.com/arana-db/arana/pull/685](https://github.com/arana-db/arana/pull/685)[https://github.com/arana-db/arana/pull/695](https://github.com/arana-db/arana/pull/695) 提交新 方案三 代码，整体执行做排序
 |  |
| [SQL] Support show table rules from xxx | @zhangbinbin | 本周请假 | [https://github.com/arana-db/arana/issues/628](https://github.com/arana-db/arana/issues/628) |
| [Bug]Optimize show open tables xxx statment. | @huxiaohan | | [https://github.com/arana-db/arana/issues/676](https://github.com/arana-db/arana/issues/676) |
| [UT] add unit test for conn_pool.go and tx.go | @dengzhengwei | [https://github.com/arana-db/arana/pull/690](https://github.com/arana-db/arana/pull/690)继续完善单测内容会后私聊 | [https://github.com/arana-db/arana/issues/686](https://github.com/arana-db/arana/issues/686) |
| [SEQUENCE] support CURRVAL,NEXTVAL for sequence in select statement | @pengjiaxin |  | [https://github.com/arana-db/arana/issues/675](https://github.com/arana-db/arana/issues/675) |

## 2. 下周工作
| 功能点 | 参与人员 | 备注 |
| --- | --- | --- |
| [Feature] add log for sharding logic SQL and real SQL. | @dongjh | [https://github.com/arana-db/arana/issues/596](https://github.com/arana-db/arana/issues/596) |
| 1、[TX] support transaction XA  @chunshao <br> ~~2、[XA] build auto clean transaction log ability~~ @pangxing ~~ <br> ~~3、[XA] build transaction log ability @liyi~~ <br> 4、[XA] build XA transaction back-and-forth decision-making  @haohongf <br> 5、[tx] add deadlock for arana transaction. @liuyucai | @chunshao <br> @liuyuecai <br> @haohongfan <br> @liyi（finish）<br> @pangxing（finish）| [https://github.com/arana-db/arana/issues/324](https://github.com/arana-db/arana/issues/324) <br> [https://github.com/arana-db/arana/issues/556](https://github.com/arana-db/arana/issues/556) <br> [https://github.com/arana-db/arana/issues/555](https://github.com/arana-db/arana/issues/555) <br> [https://github.com/arana-db/arana/issues/602](https://github.com/arana-db/arana/issues/602) <br> [https://github.com/arana-db/arana/issues/607](https://github.com/arana-db/arana/issues/607) |
| [kernel] run arana at the envoy. | @haohongfan | [https://github.com/arana-db/arana/issues/649](https://github.com/arana-db/arana/issues/649) |
| Support multiple columns complex sharding key. | @sanxun0325 @sudu | [https://github.com/arana-db/arana/issues/425](https://github.com/arana-db/arana/issues/425) |
| [Feature] Support SQL join | @sudu @huangwenkang | [https://github.com/arana-db/arana/issues/600](https://github.com/arana-db/arana/issues/600) <br> [https://github.com/arana-db/arana/issues/622](https://github.com/arana-db/arana/issues/622) |
| Implement a low-level HASH-JOIN dataset | @wangrui @sudu | [https://github.com/arana-db/arana/issues/642](https://github.com/arana-db/arana/issues/642) |
| [Bug] fix SHOW INDEXES FROM xxx result error | | [https://github.com/arana-db/arana/issues/678](https://github.com/arana-db/arana/issues/678) |
| ~~[Bug] select from table not exist cause panic~~ | @gongna | [https://github.com/arana-db/arana/issues/660](https://github.com/arana-db/arana/issues/660) |
| [Bug] fix show tables xxx statement error.  | @zhangxinyu | [https://github.com/arana-db/arana/issues/677](https://github.com/arana-db/arana/issues/677) |
| ~~[CI] Integrated code check tools~~ | @weideman | [https://github.com/arana-db/arana/issues/670](https://github.com/arana-db/arana/issues/670) |
| MySQL  window 函数实现 | @yangwei | [https://github.com/arana-db/arana/issues/667](https://github.com/arana-db/arana/issues/667) |
| [SQL] Support MySQL Prepare SQL | @maronghe | [https://github.com/arana-db/arana/issues/591](https://github.com/arana-db/arana/issues/591) |
| [SQL] Explain | @maorui |  |
| [SQL] Support show table rules from xxx | @sanxun0325 | [https://github.com/arana-db/arana/issues/628](https://github.com/arana-db/arana/issues/628) |
| [Bug]Optimize show open tables xxx statment. | @huxiaohan | [https://github.com/arana-db/arana/issues/676](https://github.com/arana-db/arana/issues/676) |
| [UT] add unit test for conn_pool.go and tx.go | @dengzhengwei | [https://github.com/arana-db/arana/issues/686](https://github.com/arana-db/arana/issues/686) |
| [SEQUENCE] support CURRVAL,NEXTVAL for sequence in select statement | @pengjiax | [https://github.com/arana-db/arana/issues/675](https://github.com/arana-db/arana/issues/675) |

### 0.2.0 规划

[https://arana.yuque.com/staff-guqen5/howkif/zr5od2](https://arana.yuque.com/staff-guqen5/howkif/zr5od2)

**Comment by maronghe** [comment]
0.2.0 can release

**Comment by maronghe** [comment]
@all 多补充一下 UT；

**Comment by dongzl** [comment]
1、事务问题月才沟通进度；
2、范哥 run envoy 去掉；
3、黄文康 PR review 建议；
4、魏德曼、张心雨、胡潇晗、王瑞
5、语雀文章内容贴到github；
